yt_dlp/extractor/wrestleuniverse.py

```diff
@@ -0,0 +1,231 @@
+import base64
+import binascii
+import json
+import time
+
+from .common import InfoExtractor
+from ..utils import (
+    ExtractorError,
+    int_or_none,
+    jwt_decode_hs256,
+    traverse_obj,
+    url_or_none,
+)
+
+
+class WrestleUniverseBaseIE(InfoExtractor):
+    _VALID_URL_TMPL = r'https?://(?:www\.)?wrestle-universe\.com/(?:(?P<lang>\w{2})/)?%s/(?P<id>\w+)'
+    _API_PATH = None
+    _TOKEN = None
+    _TOKEN_EXPIRY = None
+
+    def _get_token_cookie(self):
+        if not self._TOKEN or not self._TOKEN_EXPIRY:
+            token_cookie = self._get_cookies('https://www.wrestle-universe.com/').get('token')
+            if not token_cookie or not token_cookie.value:
+                self.raise_login_required(method='cookies')
+            self._TOKEN = token_cookie.value
+            expiry = traverse_obj(jwt_decode_hs256(self._TOKEN), ('exp', {int_or_none}))
+            if not expiry:
+                raise ExtractorError('There was a problem with the token cookie')
+            self._TOKEN_EXPIRY = expiry
```

Comment:
Why is this saved? It doesn't seem to be used elsewhere

Reply:
It is used each time the `_get_token_cookie` method is called, immediately afterwards (`if self._TOKEN_EXPIRY <= int(time.time())`). We need to check `_TOKEN_EXPIRY` each time we use the token to not send an expired token. If we dont do that inside the method, we would have to duplicate the expiration logic. This is also a reason to not do this in `_real_initialize`.